src/network/MissionControlProtocol.cpp

```diff
@@ -138,14 +139,33 @@ void MissionControlProtocol::handleSetArmIKEnabled(const json& j) {
 		}
 		DataPoint<navtypes::Vectord<Constants::arm::IK_MOTORS.size()>> armJointPositions =
 			robot::getMotorPositionsRad(Constants::arm::IK_MOTORS);
+
 		// TODO: there should be a better way of handling invalid data than crashing.
 		// It should somehow just not enable IK, but then it needs to communicate back to MC
 		// that IK wasn't enabled?
 		assert(armJointPositions.isValid());
-		Globals::planarArmController.set_setpoint(armJointPositions.getData());
-		Globals::armIKEnabled = true;
-		_arm_ik_repeat_thread =
-			std::thread(&MissionControlProtocol::updateArmIKRepeatTask, this);
+
+		// Reject enabling IK if EE outside of safety radius.
+		// Compute the new EE position to determine if it is within safety factor.
+		// If so, then enable IK by emplacing Globals::planarArmController and setting the
+		// setpoint. Otherwise, is a no-op.
+		double eeRadius =
+			Globals::planarArmKinematics.jointPosToEEPos(armJointPositions.getData()).norm();
+		double maxRadius =
+			Globals::planarArmKinematics.getSegLens().sum() * Constants::arm::SAFETY_FACTOR;
+
+		if (eeRadius <= maxRadius) {
+			if (!Globals::planarArmController.has_value()) {
+				Globals::planarArmController.emplace(navtypes::Vectord<2>(0, 0),
+													 Globals::planarArmKinematics,
+													 Constants::arm::SAFETY_FACTOR);
+			}
+
+			Globals::planarArmController.value().set_setpoint(armJointPositions.getData());
```

Comment:
If we do what I mentioned in the other review comment, we should add a method called `is_setpoint_valid()` or something to `PlanarArmController`, so we can check if the setpoint is valid before setting it, and reject otherwise.